packages/react-composites/src/composites/CallComposite/components/buttons/Reaction.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* @conditional-compile-remove(reaction) */
+import { ControlBarButtonStyles, ReactionButton, ReactionButtonProps } from '@internal/react-components';
+/* @conditional-compile-remove(reaction) */
+import React, { useMemo } from 'react';
+/* @conditional-compile-remove(reaction) */
+import { CallControlDisplayType } from '../../../common/types/CommonCallControlOptions';
+/* @conditional-compile-remove(reaction) */
+import { usePropsFor } from '../../hooks/usePropsFor';
+/* @conditional-compile-remove(reaction) */
+import { concatButtonBaseStyles } from '../../styles/Buttons.styles';
+/* @conditional-compile-remove(reaction) */
+import { _isInLobbyOrConnecting } from '@internal/calling-component-bindings';
+/* @conditional-compile-remove(reaction) */
+import { getCallStatus } from '../../selectors/baseSelectors';
+/* @conditional-compile-remove(reaction) */
+import { useSelector } from '../../hooks/useSelector';
+
+/* @conditional-compile-remove(reaction) */
+/** @private */
+export const Reaction = (props: {
+  // The value of `CallControlOptions.reactionButton`.
+  option?: boolean | { disabled: boolean };
+  displayType?: CallControlDisplayType;
+  styles?: ControlBarButtonStyles;
+  disabled?: boolean;
+}): JSX.Element => {
+  const reactionButtonProps = usePropsFor(ReactionButton) as unknown as ReactionButtonProps;
+  const callStatus = useSelector(getCallStatus);
+  const styles = useMemo(() => concatButtonBaseStyles(props.styles ?? {}), [props.styles]);
+
+  let reactionButtonDisabled = isDisabled(props.option);
+
+  if (_isInLobbyOrConnecting(callStatus)) {
```

Comment:
Is this needed? Looks like the selector is handling this

Reply:
Well, the selector was not handling the lobby one. But with the refactoring, can only disable through selector. Hence removing this method. Thanks :)